presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerification.java

```diff
@@ -59,9 +68,11 @@ public DataVerification(
             VerifierConfig verifierConfig,
             TypeManager typeManager,
             ChecksumValidator checksumValidator,
-            ListeningExecutorService executor)
+            ListeningExecutorService executor,
+            SnapshotQueryConsumer snapshotQueryConsumer,
+            Map<String, SnapshotQuery> snapshotQueries)
     {
-        super(queryActions, sourceQuery, exceptionClassifier, verificationContext, Optional.empty(), verifierConfig, executor);
+        super(queryActions, sourceQuery, exceptionClassifier, verificationContext, Optional.empty(), verifierConfig, executor, snapshotQueryConsumer, snapshotQueries);
```

Comment:
Instead of adding the snapshot logic here have you considered creating a new class `SnapshotDataVerification` which extends DataVerification . That way you can keep the logic a bit cleaner in a Object Oriented fashion and control the injection through `VerificationFactory` .

Also this will help get rid of `protected`  variables in AbstractVerification class you added.